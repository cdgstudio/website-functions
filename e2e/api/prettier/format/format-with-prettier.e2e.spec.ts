import { describe, expect, it } from '@jest/globals';

describe('/api/code/format', () => {
  it('should format typescript code', async () => {
    const query = new URLSearchParams({
      language: 'typescript',
    });

    const response = await fetch(new URL(`http://localhost:8888/api/code/format?${query}`), {
      method: 'POST',
      body: `class A{
        constructor()  {  }
        }`,
    });

    const text = await response.text();

    expect(text).toMatchSnapshot();
  });

  it('should format XML code', async () => {
    const query = new URLSearchParams({
      language: 'xml',
    });

    const response = await fetch(new URL(`http://localhost:8888/api/code/format?${query}`), {
      method: 'POST',
      body: `<button appTooltip="Hello from string">Tooltip as string</button>

      <svg [appTooltip]="templateForTooltip" >
        <!-- ... -->
      </svg>
      <ng-template #templateForTooltip> <strong>Hello</strong> from template </ng-template>`,
    });

    const text = await response.text();

    expect(text).toMatchSnapshot();
  });
});
