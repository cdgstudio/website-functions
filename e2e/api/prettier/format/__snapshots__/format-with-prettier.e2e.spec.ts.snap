// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/api/code/format should format XML code 1`] = `
"<button appTooltip="Hello from string">Tooltip as string</button>

<svg [appTooltip]="templateForTooltip">
  <!-- ... -->
</svg>
<ng-template #templateForTooltip> <strong>Hello</strong> from template </ng-template>
"
`;

exports[`/api/code/format should format typescript code 1`] = `
"class A {
  constructor() {}
}
"
`;
